pkg/sql/pgwire/types.go

```diff
@@ -116,7 +120,11 @@ func (b *writeBuffer) writeTextDatum(
 		b.writeLengthPrefixedString(v.IPAddr.String())
 
 	case *tree.DString:
-		b.writeLengthPrefixedString(string(*v))
+		sv := string(*v)
+		if T.Oid() == oid.T_bpchar {
```

Comment:
Just checking, bpchar is the opposite of varchar right (using fixed width)? If so, this needs a comment, and so does the formatting directive here (idk what this does)